File: ProjectEuler11/ProjectEuler11.py
def main():
	with open("infile.txt") as file:
		data = file.read()
	
	maximum = 0																												#to store the max product
	data = data.split('\n')																									#split the data according to line separation
	numbers = []
	
	for values in data:
		numbers.append(values.split(' '))																					#split the line with respect to space and hence create a 2D list
	
	for i in numbers:																										#to get the maximum horizontal product
		for j in range(0, len(i) - 4):
			product = int(i[j]) * int(i[j+1]) * int(i[j+2]) * int(i[j+3])
			if product > maximum:
				maximum = product
	
	for i in range (0, len(numbers) - 4):																					#to get the maximum vertical product
		for j in range(0, len(numbers[i])):
			product = int(numbers[i][j]) * int(numbers[i+1][0]) * int(numbers[i+2][j]) * int(numbers[i+3][j])
			if product > maximum:
				maximum = product

	for i in range(0, len(numbers) - 4):																					#to get the maximum diagonal(positive direction) product
		for j in range(0, len(numbers[i]) - 4):
			product = int(numbers[i][j]) * int(numbers[i+1][j+1]) * int(numbers[i+2][j+2]) * int(numbers[i+3][j+3])
			if product > maximum:
				maximum = product

	for i in range(0, len(numbers) - 4):																					#to get the maximum diagonal(negative direction) product
		for j in range(len(numbers[i]) - 1, 3, -1):
			product = int(numbers[i][j]) * int(numbers[i+1][j-1]) * int(numbers[i+2][j-2]) * int(numbers[i+3][j-3])
			if product > maximum:
				maximum = product

	return maximum

print(main())

File: ProjectEuler11/ProjectEuler11.java
/*THIS PROJECT, I START WITH 1 AS MY POSITION RATHER THAN 0 DUE TO A SMALL ERROR THAT I CAN'T FIGURE OUT. WHEN I READ FROM THE FILE, IT IS AUTOMATICALLY LEAVING ONE WHOLE LINE IN ARRAY AS BLANK. IF SOMEONE KNOWS WHAT I'M DOING WRONG, PLEASE DROP AN EMAIL AT aaryabc49@gmail.com */

import java.io.*;
import java.util.Scanner;

public class ProjectEuler11{

	public int[][] numberList() throws IOException{
		int[][] grid = new int[21][20];
		int i = 0;
		int j = 0;
		Scanner scanner = new Scanner(new File("infile.txt"));
		while (scanner.hasNextInt()){
			if (j % 20 == 0){
				i += 1;
				j = 0;
			}
			grid[i][j] = scanner.nextInt();
			j += 1;
		}
		return grid;
	}
	
	public static void main(String[] args) throws IOException{
		ProjectEuler11 call = new ProjectEuler11();
		long max = 0, product = 0;
		int[][] grid = call.numberList();
		
		for (int i = 1; i < 21; i++){																		//for horizontal line
			for (int j = 0; j < 17; j++){
				product = grid[i][j] * grid[i][j+1] * grid[i][j+2] * grid[i][j+3];
				if (product > max)
					max = product;
			}
		}
		
		for (int j = 0; j < 20; j++){																		//for vertical line
			for (int i = 1; i < 18; i ++){
				product = grid[i][j] * grid[i+1][j] * grid[i+2][j] * grid[i+3][j];
				if (product > max)
					max = product;
			}
		}

		for (int i = 1; i < 18; i++){																		//for positive direction diagonal
			for (int j = 0; j < 17; j++){
				product = grid[i][j] * grid[i+1][j+1] * grid[i+2][j+2] * grid[i+3][j+3];
				if (product > max)
					max = product;
			}
		}

		for (int i = 1; i < 18; i++){																		//for negative direction diagonal
			for (int j = 19; j > 2; j--){
				product = grid[i][j] * grid[i+1][j-1] * grid[i+2][j-2] * grid[i+3][j-3];
				if (product > max)
					max = product;
			}
		}

		System.out.println(max);
	}
}